### ISSUE TITLE
`TSESLint.RuleTester` doesn't accept customization of its `describe` and `it`.

### ISSUE THREAD

**Issue by yudai-nkt** [typescript-eslint/typescript-eslint#4422]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

ESLint's `RuleTester` supports [behavior customization](https://eslint.org/docs/developer-guide/nodejs-api#customizing-ruletester) via its static methods (i.e., `RuleTester.describe` and `RuleTester.it`), but it looks like `TSESLint.RuleTester` does not. I'd appreciate it if `TSESLint.RuleTester` supports this feature as well.

**Repro**

```js
const { TSESLint } = require("@typescript-eslint/experimental-utils");

const rule = (context) => ({});
const ruleTester = new TSESLint.RuleTester();

TSESLint.RuleTester.it = (text, callback) => {
  console.log("RuleTester has been customized!")
}

ruleTester.run("repro", rule, {
  valid: ["var foo = 0;"],
  invalid: [
    {
      code: "var foo = 0;",
      errors: [{ message: "This is actually valid, so the test should fail." }],
    },
  ],
});
```

**Expected Result**

`RuleTester has been customized!` appears on the terminal twice and the test is not run (because `callback` is not called in the customization).

**Actual Result**

`RuleTester has been customized!` does not show up, and the test is run.

```console
$ node repro.js
/Users/ynkt/src/github.com/yudai-nkt/eslint-ruletester-uvu/node_modules/eslint/lib/rule-tester/rule-tester.js:332
        throw err;
        ^

AssertionError [ERR_ASSERTION]: Should have 1 error but had 0: []
    at testInvalidTemplate (/Users/ynkt/src/github.com/yudai-nkt/eslint-ruletester-uvu/node_modules/eslint/lib/rule-tester/rule-tester.js:773:24)
    at Function.<anonymous> (/Users/ynkt/src/github.com/yudai-nkt/eslint-ruletester-uvu/node_modules/eslint/lib/rule-tester/rule-tester.js:997:29)
    at Function.itDefaultHandler (/Users/ynkt/src/github.com/yudai-nkt/eslint-ruletester-uvu/node_modules/eslint/lib/rule-tester/rule-tester.js:327:23)
    at /Users/ynkt/src/github.com/yudai-nkt/eslint-ruletester-uvu/node_modules/eslint/lib/rule-tester/rule-tester.js:994:63
    at Array.forEach (<anonymous>)
    at Function.<anonymous> (/Users/ynkt/src/github.com/yudai-nkt/eslint-ruletester-uvu/node_modules/eslint/lib/rule-tester/rule-tester.js:993:30)
    at Function.describeDefaultHandler (/Users/ynkt/src/github.com/yudai-nkt/eslint-ruletester-uvu/node_modules/eslint/lib/rule-tester/rule-tester.js:344:19)
    at Function.<anonymous> (/Users/ynkt/src/github.com/yudai-nkt/eslint-ruletester-uvu/node_modules/eslint/lib/rule-tester/rule-tester.js:992:24)
    at Function.describeDefaultHandler (/Users/ynkt/src/github.com/yudai-nkt/eslint-ruletester-uvu/node_modules/eslint/lib/rule-tester/rule-tester.js:344:19)
    at RuleTester.run (/Users/ynkt/src/github.com/yudai-nkt/eslint-ruletester-uvu/node_modules/eslint/lib/rule-tester/rule-tester.js:980:20) {
  generatedMessage: false,
  code: 'ERR_ASSERTION',
  actual: 0,
  expected: 1,
  operator: 'strictEqual'
}
```

**Additional Info**

If I use ESLint's `RuleTester` instead, the result is what I expected.

```js
const { RuleTester } = require("eslint")

const rule = (context) => ({});
const ruleTester = new RuleTester();

RuleTester.it = (text, callback) => {
  console.log("RuleTester has been customized!")
}

ruleTester.run("repro", rule, {
  valid: ["var foo = 0;"],
  invalid: [
    {
      code: "var foo = 0;",
      errors: [{ message: "This is actually valid, so the test should fail." }],
    },
  ],
});
```

```console
$ node repro.js
RuleTester has been customized!
RuleTester has been customized!
```

**Versions**

| package                                 | version |
| --------------------------------------- | ------- |
| `@typescript-eslint/experimental-utils` | `5.9.0` |
| `TypeScript`                            | N/A |
| `node`                                  | `16.13.1` |

**Comment by yudai-nkt** [comment]
I took a look at the source code, but all this package do is just to annotate types. I can't figure out why the reproductio above doesn't work...

https://github.com/typescript-eslint/typescript-eslint/blob/97c0e8606705dc0b4ef8b4ee5206111276c8a170/packages/experimental-utils/src/ts-eslint/RuleTester.ts#L139-L187

**Comment by bradzacher** [comment]
Out of curiosity - what's your usecase for customising the `it`/`describe`?

**Comment by yudai-nkt** [comment]
I've recently started developing an ESLint plugin for a test runner called [`uvu`](https://github.com/lukeed/uvu) (note that it is not TypeScript-specific; I just want to write the plugin in TypeScript). Since `uvu` does not provide global `it`/`describe`, I need the customization.

Below is a close-to-minimal and real-world illustration of what I want to achieve with `TSESLint.RuleTester`.

```javascript
const { RuleTester } = require("eslint")
const { suite } = require("uvu")

const rule = (context) => ({});
const ruleTester = new RuleTester();

const Test = suite("A test suite.")

RuleTester.it = (text, callback) => {
  return Test(text, callback)
}

ruleTester.run("repro", rule, {
  valid: ["var foo = 0;"],
  invalid: [
    {
      code: "var foo = 0;",
      errors: [{ message: "This is actually valid, so the test should fail." }],
    },
  ],
});

Test.run()
```

```console
$ node test.js
 A test suite.  • ✘   (1 / 2)

   FAIL  A test suite.  "var foo = 0;"
    Should have 1 error but had 0: []  (strictEqual)

        ++1    (Expected)
        --0    (Actual)

    at testInvalidTemplate (/Users/ynkt/src/github.com/yudai-nkt/eslint-ruletester-uvu/node_modules/eslint/lib/rule-tester/rule-tester.js:773:24)
    at Object.handler (/Users/ynkt/src/github.com/yudai-nkt/eslint-ruletester-uvu/node_modules/eslint/lib/rule-tester/rule-tester.js:997:29)
    at Number.runner (/Users/ynkt/src/github.com/yudai-nkt/eslint-ruletester-uvu/node_modules/uvu/dist/index.js:77:16)
    at async Timeout.exec [as _onTimeout] (/Users/ynkt/src/github.com/yudai-nkt/eslint-ruletester-uvu/node_modules/uvu/dist/index.js:138:33)



  Total:     2
  Passed:    1
  Skipped:   0
  Duration:  22.85ms
```

Of course I can use other test runners like Jest or Mocha, but I want to use the test runner which I'm writing a plugin for.

**Comment by bradzacher** [comment]
So this is a bug in ESLint:
https://github.com/eslint/eslint/blob/03ac8cfc773279c01a62897692160f9a883ff4f5/lib/rule-tester/rule-tester.js#L980-L1002

ESLint uses `RuleTester.describe` et al which directly references the  `RuleTester` class, ignoring whatever value is set on the subclass.

**Comment by yudai-nkt** [comment]
I didn't notice that this was an upstream issue, thank you so much for your investigation and PR!
